bluebottle/time_based/effects.py

```diff
@@ -197,3 +197,19 @@ def post_save(self, **kwargs):
         if activity.slot_selection == 'all':
             for slot in activity.slots.all():
                 SlotParticipant.objects.create(participant=participant, slot=slot)
+
+
+class CreateSlotParticipantsForParticipantsEffect(Effect):
+    title = _('Add participants to all slots if slot selection is set to "all"')
+    template = 'admin/create_slot_participants_for_participants.html'
+
+    @property
+    def display(self):
+        return self.instance.activity.slot_selection == 'all'
+
+    def post_save(self, **kwargs):
+        slot = self.instance
+        activity = self.instance.activity
+        if activity.slot_selection == 'all':
+            for participant in activity.accepted_participants:
```

Comment:
I think we now create slot_participants on participant.initiate, so we should either move that to participant.approve or add slot_participants for all acitivity.participants.
In the first case we should also have some logic to the participant.approve/restore that loops over the slots and do a get_or_create or something similar.

Reply:
Yeah, you are right. I wanted to fix that but forgot about that. There is a failing test for this though.